nomad/structs/structs.go

```diff
@@ -6432,12 +6447,67 @@ func (a *AllocMetric) ExhaustQuota(dimensions []string) {
 	a.QuotaExhausted = append(a.QuotaExhausted, dimensions...)
 }
 
+// ScoreNode is used to gather top K scoring nodes in a heap
 func (a *AllocMetric) ScoreNode(node *Node, name string, score float64) {
-	if a.Scores == nil {
-		a.Scores = make(map[string]float64)
+	if a.topScores == nil {
+		a.topScores = make(map[string]*lib.ScoreHeap)
+	}
+	scorerHeap, ok := a.topScores[name]
```

Comment:
I do not believe this is the API we want. It is possible to not have all the sub scores for a the top k normalized scores.